index.js

```diff
@@ -2,527 +2,533 @@ import { NativeModules } from 'react-native';
+     * Audience Builder “Purchase Order Var Is” query. For example, you could specify the shipping
+     * address, estimated delivery date given, credit card type used, whether a deal was used, the
+     * promo code used, etc.Note that a vars object may also exist at the item level. See the example
+     * code, and the items definition in this table.You may use any custom order variable name(s).
+     * Note that the following variable name is reserved at the order level, for a particular purpose:
+     *     st_cost – The client’s cost for the items in the purchase, in cents. The value should be
+     *     an integer, and is recommended if you are using Retention Analytics in order to report
+     *     net revenue.
+     * @param {Object} vars map containing the custom fields for the purchase.
+     */
+    this.setVars = function(vars) {
+      this.vars = vars;
+    }
+
+    /**
+     * Set the message ID associated with the purchase. This is required to have revenue data matched
+     * to email sends in Sailthru. Pass the identifying message_id of the email the user is coming from;
+     * this will be the value stored in the sailthru_bid cookie for your domain. The message attribution
+     * will be displayed in your Campaign Summary, Transactional Report, Purchase Log, and in Lifecycle
+     * Optimizer Metrics.
+     * @param {string} messageId the message ID
+     */
+    this.setMessageId = function(messageId) {
+      this.message_id = messageId;
+    }
   }
 
   /**
-   * Set the message ID associated with the purchase. This is required to have revenue data matched
-   * to email sends in Sailthru. Pass the identifying message_id of the email the user is coming from;
-   * this will be the value stored in the sailthru_bid cookie for your domain. The message attribution
-   * will be displayed in your Campaign Summary, Transactional Report, Purchase Log, and in Lifecycle
-   * Optimizer Metrics.
-   * @param {string} messageId the message ID
+   * Creates a Purchase object from an array of ContentItem objects.
+   * @param {Array} contentItems an array of ContentItem objects.
    */
-  this.setMessageId = function(messageId) {
-    this.message_id = messageId;
+  SailthruMobile.Purchase.fromContentItems = function(contentItems) {
+    var purchaseItems = [];
+    contentItems.forEach(function (item, index) {
+      purchaseItems.push(new Carnival.PurchaseItem(item));
+    });
+    return new this(purchaseItems);
   }
-}
 
-/**
- * Creates a Purchase object from an array of ContentItem objects.
- * @param {Array} contentItems an array of ContentItem objects.
- */
-SailthruMobile.Purchase.fromContentItems = function(contentItems) {
-  var purchaseItems = [];
-  contentItems.forEach(function (item, index) {
-    purchaseItems.push(new Carnival.PurchaseItem(item));
-  });
-  return new this(purchaseItems);
+  module.exports = SailthruMobile;
 }
```

Comment:
Only thing I would suggest is to have `module.exports = ...` in one place.

So, rather than having it in each block

```
if (Carnival) {
  ...
  module.exports = Carnival;
}

...

if (SailthruMobile) {
  ...
  module.exports = SailthruMobile;
}
```

you could have it at the bottom of the file and use a ternary instead. eg. 

```
  module.exports = Carnival ? Carnival : SailthruMobile;
```

Reply:
👍